### PR TITLE
Return group roles as part of /users/me response.

### PR DESCRIPTION
**Pull request by jgadling**
### Summary:
- **What:** Add groups & group roles to `/v2/users/me` response.
- **Ticket:** [sc192742](https://app.shortcut.com/genepi/story/192742)
- **Env:** `<rdev link>`

### Notes:
This builds on #1184

### Demos:
The response format looks like this for now, we can tweak it as necessary if it helps with anything @mmmmmmaya 
Screen Shot 2022-06-24 at 11 11 17 AM

- [ ] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

**Comment by mmmmmmaya** [comment]
I assume this is right but confirming top level `group` = currently authed group and the `groups` is just all group user belongs to?

this is 😻

**Comment by jgadling** [comment]
> I assume this is right but confirming top level `group` = currently authed group and the `groups` is just all group user belongs to?
> 
> this is 😻

We can do this two ways.

First way (my preference): The top level group and group_admin flag are going away once we finish this migration. A user will always have access to all their groups, and it's up to the front end to send the current group context to the backend when asking for samples/trees/etc. 

Second way, also will work fine but seems a little weird to me?: The frontend is welcome to send a group ID to the `/users/me` endpoint (it'll be like `/v2/orgs/1234/users/me` and I can populate the top level group and group admin fields with the group referenced in the URL.

**Comment by mmmmmmaya** [comment]
The first way you proposed sounds good to me!

**Review by phoenixAja — APPROVED** [review]
looks good to me!

File: src/backend/aspen/test_infra/models/usergroup.py [hunk 1]
```diff
@@ -47,7 +50,7 @@ def user_factory(
     system_admin=False,
     agreed_to_tos=True,
 ) -> User:
-    return User(
+    user = User(
```
Comment (lvreynoso) [inline src/backend/aspen/test_infra/models/usergroup.py:53]: Will this function still be used after this PR, or is not every instance replaced with `userrole`? Could it be deleted?

**Review by lvreynoso — APPROVED** [review]
lgtm

File: src/backend/aspen/test_infra/models/usergroup.py (hunk 1 above)
Comment (jgadling) [reply src/backend/aspen/test_infra/models/usergroup.py:53]: Ohhh good point! our workflow tests still use this method but I think we can work on removing it as part of our oso migration.

**Pull request merged**